File: scalechain-storage/src/test/scala/io/scalechain/blockchain/storage/index/BlockDatabaseWithRocks.scala
package io.scalechain.blockchain.storage.index


import java.io.File

import io.scalechain.blockchain.storage.Storage
import org.apache.commons.io.FileUtils
import org.scalatest._

/**
  * Created by kangmo on 11/2/15.
  */
class BlockDatabaseWithRocks extends BlockDatabaseTestTrait with BeforeAndAfterEach {
  this: Suite =>

  Storage.initialize()

  var db : BlockDatabase = null

  override def beforeEach() {

    val testPath = new File("./target/unittests-BlockDatabaseWithRocks")
    FileUtils.deleteDirectory( testPath )
    db = new BlockDatabase( new RocksDatabase( testPath ) )

    super.beforeEach()
  }

  override def afterEach() {
    super.afterEach()

    db.close()
  }
}

File: scalechain-storage/src/test/kotlin/io/scalechain/blockchain/storage/index/BlockDatabaseWithRocks.kt
package io.scalechain.blockchain.storage.index

import io.kotlintest.KTestJUnitRunner
import java.io.File
import io.scalechain.blockchain.storage.Storage
import org.junit.runner.RunWith

/**
  * Created by kangmo on 11/2/15.
  */
@RunWith(KTestJUnitRunner::class)
class BlockDatabaseWithRocks : BlockDatabaseTestTrait() {

  init {
    Storage.initialize()
    addTests()
  }

  val testPath = File("./build/unittests-BlockDatabaseWithRocks")


  lateinit override var db : KeyValueDatabase
  lateinit override var blockDb : BlockDatabase

  override fun beforeEach() {
    testPath.deleteRecursively()
    testPath.mkdir()

    db = DatabaseFactory.create(testPath)
    blockDb = object : BlockDatabase {}


    super.beforeEach()
  }

  override fun afterEach() {
    super.afterEach()

    db.close()
  }
}
